src/core/cgroup.c

```diff
@@ -3608,6 +3608,9 @@ int unit_cgroup_freezer_action(Unit *u, FreezerAction action) {
         assert(u);
         assert(IN_SET(action, FREEZER_FREEZE, FREEZER_THAW));
 
+        if (!cg_freezer_supported())
+                return 0;
```

Comment:
shouldn't this fail with EOPNOTSUPP? And the caller should catch and possibly ignore the error?

Reply:
We handle it like this in routine on DBus API level (`unit_freezer_action()`). I think this low-level function doesn't need to do that. It should be called in almost all the cases from context where we already established that freezer is supported. The call from job engine just before we apply the job is best-effort thing and we doesn't even check the return value.

Reply:
i see...

maybe we should one day rename unit_cgroup_freezer_action() → unit_freezer_action_lowlevel() or so, to clarify that it is a low-level version of some function. Too easy to confuse with unit_freezer_action() which also exists, but is the high-level op. I for one got this confused...